### ISSUE TITLE
checkbox: inside menu-item does not get focus using keyboard navigation

### ISSUE THREAD

**Issue by ChaniFruchtman** [angular/material#12051]
when checkbox is added inside a menu, it's not considered as a menu item. it's not getting focus as any other menu-item (using the up-down arrows is supposed to navigate between menu-items. the checkbox is skipped)

codepen: https://codepen.io/chanifruchtman/pen/JjRKNpx 

Reproduction Steps:
1. Create a `md-menu` element, and put a `md-checkbox` inside `md-menu-item`
2. Open the menu, use arrow keys for navigation.

expected behavior:
Checkbox element get focused 

Current behavior:
Keyboard navigation escapes the checkbox.

Motivation for change:
Users of keyboard navigation should be able to press all menu items.

AngularJS: 1.8.2
AngularJS Material: 1.2.1

**Comment by Splaktar** [comment]
Thank you for taking the time to submit this issue. However it does not follow our **required** issue template. 

This gives the team the information needed to investigate the bug.

**Comment by Splaktar** [comment]
Thank you for the updated reproduction steps and additional information.

It looks like the checkbox gets the initial focus if you use the keyboard to open the menu (space bar). But then if you go down to the other options, it won't let you go back up to focus the checkbox again.

If you open the menu with the mouse, the checkbox gets focus but there is no visual indication that it is focused.

**Comment by Splaktar** [comment]
From the [menu autofocus docs](https://material.angularjs.org/latest/api/directive/mdMenu#auto-focus):

> By default, when a menu opens, `md-menu` focuses the first button in the menu content.

We don't support `md-checkbox` directly inside of a `md-menu-item` (there has to be a `md-button` to get the built in focus handling). However, if you are careful about the event handling, you can put a `md-checkbox` inside of a `md-button` that is inside of a `md-menu-item`. [Working Demo](https://codepen.io/Splaktar/pen/zYKNLpa)

Also note that the [`md-menu-bar`](https://material.angularjs.org/latest/api/directive/mdMenuBar) has support for `md-menu-items` with types like `checkbox` and `radio`.
